search/elastic.py

```diff
@@ -709,6 +716,25 @@ def search(self,
         _sort_args_in_body = kwargs.pop('sort', None)
         if _sort_args_in_body:
             body['sort'] = _sort_args_in_body
+        # Get `doc_count` from ES ( without filters )
+        # E.g: if we query courses with lots of conditions, then we still return total count of
+        # courses with this Flag `ga_total`
+        ga_total = kwargs.pop('ga_total', None)
+        if ga_total:
+            body["aggs"] = {
+                "total_records": {
+                    "global": {}
+                }
+            }
+            ### For low level Roles:
+            if field_dictionary.get("course", None):    # Courses
+                body["aggs"]["total_records"]["aggs"] = {
+                    "filtered_id": {"filter": {"terms": {"_id": field_dictionary["course"]}}}
```

Comment:
- Global Aggregation on `courses/programs IDs` if they're specified. Then we can get the correct total number for different Role Level